Automattic/jetpack
# Podcast Player: refactor player interactions into a class

Refactor player handling from view.js into a self-contained class `PodcastPlayer`.

#### Changes proposed in this Pull Request:
* move player behavior from view.js where it was handled by several functions and objects holding state

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
paYJgx-sH-p2

#### Testing instructions:
- Rebuild extensions
- Insert a player block (or more) into a post and continue testing in the front end preview
- All operations of the player should work as you expect.
- Playing/pausing a track by:
  - Play/Pause button in the player
  - Clicking an episode link (however, clicking the same link with CMD modifier must do nothing and open the episode in a new tab)
  - Navigating the focus to an episode link and
    - pressing <kbd>Enter</kbd>
    - pressing <kbd>Space</kbd>
- Icons for an active episode showing/hiding as expected
- Errors shown when accessing non-playable episode (simulate by blocking requests/going offline)
  - Errors disappearing once another episode is selected
  - Link inside an error works
- Player instances are accessible in `window.jetpackPodcastPlayers.playerInstances` - you can control it externally by using its public methods `.play()`, `.pause()`, `.togglePlayPause()` and `.selectTrack( i )` where `i` is an index of episode in the list. Counted from zero as we developers love.
- Run `localStorage.debug = '*'` and reload the preview. You should see a message in the console about constructing a player instance.

#### Sub-issues to check if addressed:
- Clicking a Podcast Episode Button should immediately apply the is-active and aria-pressed states instead of waiting for the track to load.
- Switching episodes should use `player.tracks` to get audio URL. Attribute `data-jetpack-podcast-audio` should be completely removed. 
- Podcast Player: Add `debug` and use it to report player initialization

#### Proposed changelog entry for your changes:
* none, WIP

Closes: #15038
